pkg/core/resources/apis/mesh/fault_injection_validator.go

```diff
@@ -0,0 +1,128 @@
+package mesh
+
+import (
+	"net/http"
+	"reflect"
+	"regexp"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	"github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/validators"
+)
+
+func (f *FaultInjectionResource) Validate() error {
+	var err validators.ValidationError
+	err.Add(f.validateSources())
+	err.Add(f.validateDestinations())
+	err.Add(f.validateConf())
+	return err.OrNil()
+}
+
+func (f *FaultInjectionResource) HasFaultDelay() bool {
+	faultDelay := f.Spec.Conf.GetDelay()
+	return faultDelay != nil && !reflect.DeepEqual(*faultDelay, v1alpha1.FaultInjection_Conf_Delay{})
+}
+
+func (f *FaultInjectionResource) HasFaultAbort() bool {
+	faultAbort := f.Spec.Conf.GetAbort()
+	return faultAbort != nil && !reflect.DeepEqual(*faultAbort, v1alpha1.FaultInjection_Conf_Abort{})
+}
+
+func (f *FaultInjectionResource) HasFaultResponseBandwidth() bool {
+	faultResponseBandwidth := f.Spec.Conf.GetResponseBandwidth()
+	return faultResponseBandwidth != nil && !reflect.DeepEqual(*faultResponseBandwidth, v1alpha1.FaultInjection_Conf_ResponseBandwidth{})
+}
+
+func (f *FaultInjectionResource) validateSources() validators.ValidationError {
```

Comment:
Marco wanted to add `protocol: http` requirement to both sources and destinations

Reply:
Added additional validator, please take a look